### PR TITLE
Migrating to null safety

### LINKED ISSUE #176: Null safety support
## 🚀 Feature Requests

See https://dart.dev/null-safety/migration-guide

### Contextualize the feature

Package authors are being encouraged to start migrating their packages by creating a null-safety preview release, starting at the leaves of the dependency graph. For example, flutter_tts will need to migrate to null safety before packages that depend on flutter_tts can migrate to null safety.

### Describe the feature

https://dart.dev/null-safety
https://dart.dev/null-safety/faq

### Platforms affected (mark all that apply)
- [x] :iphone: iOS
- [x] :robot: Android

### PR DESCRIPTION
**Pull request by dlutton**
closes #176

### CODE REVIEW

File: example/lib/main.dart [hunk 1]
```diff
@@ -15,14 +15,14 @@ class MyApp extends StatefulWidget {
 enum TtsState { playing, stopped, paused, continued }
 
 class _MyAppState extends State<MyApp> {
-  FlutterTts flutterTts;
+  late FlutterTts flutterTts;
   dynamic languages;
```
Comment (jakemac53) [inline example/lib/main.dart:19]: It looks like this is required to be a `List<String>` (or at least `getLanguageDropDownMenuItems` requires this). I would give it that type if you agree with that assessment, that may require some earlier casts in other places but would overall be better/safer.

File: lib/flutter_tts_web.dart [hunk 2]
```diff
@@ -9,8 +9,8 @@ enum TtsState { playing, stopped, paused, continued }
 
 class FlutterTtsPlugin {
   static const String PLATFORM_CHANNEL = "flutter_tts";
-  static MethodChannel channel;
-  bool awaitSpeakCompletion = false;
+  static late MethodChannel channel;
+  bool? awaitSpeakCompletion = false;
```
[the remark this answers is not in the document]
Comment (jakemac53) [reply lib/flutter_tts_web.dart:13]: It is assigned a `bool?` here https://github.com/dlutton/flutter_tts/blob/f1582f7c7d82f7b5eeba60c501e641fc88f3a914/lib/flutter_tts_web.dart#L76 which could be defaulted to some value in order to remove the nullability

[the remark this answers is not in the document]
Comment (dlutton) [reply lib/flutter_tts_web.dart:13]: It might be better to leave this as is then.

[the remark this answers is not in the document]
Comment (ryanheise) [reply lib/flutter_tts_web.dart:13]: I think what Jake's suggesting is that you can trace back and fix the nullability at the source by providing a default value (alternatively if it should never be null anyway, by using the ! Postfix operator as a sort of assertion - again, at the source).

[the remark this answers is not in the document]
Comment (jakemac53) [reply lib/flutter_tts_web.dart:13]: Right, you could remove the nullability by changing like 76 that I linked to either just cast directly to `bool` if it is never null, or do `(call.arguments as bool?) ?? <some-default>`.

[the remark this answers is not in the document]
Comment (dlutton) [reply lib/flutter_tts_web.dart:13]: @jakemac53, the default as you originally commented on makes sense here.

File: example/lib/main.dart (hunk 1 above)
Comment (dlutton) [reply example/lib/main.dart]: Done

File: lib/flutter_tts_web.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (dlutton) [reply lib/flutter_tts_web.dart]: Done

**Review by jakemac53 — APPROVED** [review]

File: lib/flutter_tts_web.dart [hunk 3]
```diff
@@ -29,10 +29,10 @@ class FlutterTtsPlugin {
...
   }
 
-  html.SpeechSynthesis synth;
-  html.SpeechSynthesisUtterance utterance;
-  List<dynamic> voices;
-  List<String> languages;
+  html.SpeechSynthesis? synth;
+  late html.SpeechSynthesisUtterance utterance;
+  List<dynamic>? voices;
+  List<String?>? languages;
```
Comment (ryanheise) [inline lib/flutter_tts_web.dart:35]: This looks strange -- what would it mean for a language in the list to be null?

File: lib/flutter_tts_web.dart [hunk 4]
```diff
@@ -125,58 +110,58 @@ class FlutterTtsPlugin {
...
+  void _setLanguage(String? language) => utterance.lang = language;
 
-  bool _isLanguageAvailable(String language) {
+  bool _isLanguageAvailable(String? language) {
     if (voices?.isEmpty ?? true) _setVoices();
     if (languages?.isEmpty ?? true) _setLanguages();
-    for (var lang in languages) {
-      if (lang.toLowerCase() == language.toLowerCase()) return true;
+    for (var lang in languages!) {
+      if (lang!.toLowerCase() == language!.toLowerCase()) return true;
```
Comment (ryanheise) [inline lib/flutter_tts_web.dart:145]: The use of `lang!` here would suggest that the element type of the list probably should be non-nullable.

**Pull request merged**